Makefile.js

```diff
@@ -182,64 +182,115 @@ function getReleaseType(version) {
 }
 
 /**
+ * Gets all changes since the last tag that represents a version.
+ * @returns {Object} An object containing all the changes since the last version.
+ * @private
+ */
+function calculateReleaseInfo() {
+
+    // get most recent tag
+    var tags = getVersionTags(),
+        lastTag = tags[tags.length - 1],
+        commitFlagPattern = /([a-z]+):/i,
+        releaseInfo = {};
+
+    // get log statements
+    var logs = execSilent("git log --no-merges --pretty=format:\"* %h %s (%an)\" " + lastTag + "..HEAD").split(/\n/g);
+
+    // arrange change types into categories
+    logs.forEach(function(log) {
+        var flag = log.match(commitFlagPattern)[1].toLowerCase();
+
+        if (!releaseInfo["changelog_" + flag]) {
+            releaseInfo["changelog_" + flag] = [];
+        }
+
+        releaseInfo["changelog_" + flag].push(log);
+    });
+
+    var output = logs.join("\n"); // and join it into a string
+    releaseInfo.raw = output;
+
+    if (releaseInfo.changelog_breaking) {
```

Comment:
I can't think of a reason why we'd want to do that, but if such an occurrence happens, we could just put in a dummy commit that says "Breaking" and explains why we're doing that.